src/qt/walletmodel.cpp

```diff
@@ -1534,6 +1534,9 @@ void WalletModel::handleBip47Keys(int receiverAccountNum, void * pBlockIndex_)
             receiverAccountNumQueue.clear();
         }
         LOCK(cs_main);
+        for (; !chainActive.Contains(pBlockIndex); pBlockIndex = pBlockIndex->pprev) {
+            if (!pBlockIndex) return;
+        }
```

Comment:
you don't need to scan the entire history of blocks, if active chain contains `pBlockIndex` it contains all its ancestors

Reply:
I don't get your point. It does not scan the entire history. It goes down to the first block which is on the active chain and starts rescanning from there.

Reply:
`pBlockIndex->pprev` is an ancestor i.e. _previous_ block, not the next one. You're scanning the entire block chain up to genesis block here. Testing if `pBlockIndex` is in the chain is enough

Reply:
Or sorry, what are you trying to compute here, maybe I didn't understand. Fork point?

Reply:
There is a member of `CChain` called `FindFork` to do this

Reply:
Yes, it looks up for a fork point.